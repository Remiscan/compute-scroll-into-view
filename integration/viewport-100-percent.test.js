beforeAll(async () => {
  await page.goto('http://localhost:3000/integration/viewport-100-percent')
})

describe('scrollMode: if-needed (outside the scrollingElement bounding box)', () => {
  describe('vertical', () => {
    test('completely below the fold', async () => {
      const actual = await page.evaluate(() => {
        window.scrollTo(0, 0)
        return window
          .computeScrollIntoView(document.querySelector('.vertical-scroll .target'), {
            scrollMode: 'if-needed',
          })
          .map(window.mapActions)
      })
      expect(actual).toHaveLength(1)
      expect(actual).toMatchSnapshot()
    })

    test('partially below the fold', async () => {
      const actual = await page.evaluate(() => {
        window.scrollTo(0, 50)
        return window
          .computeScrollIntoView(document.querySelector('.vertical-scroll .target'), {
            scrollMode: 'if-needed',
          })
          .map(window.mapActions)
      })
      expect(actual).toHaveLength(1)
      expect(actual).toMatchSnapshot()
    })

    test('completely in view', async () => {
      const actual = await page.evaluate(() => {
        window.scrollTo(0, window.innerHeight / 2);
        return window
          .computeScrollIntoView(document.querySelector('.vertical-scroll .target'), {
            scrollMode: 'if-needed',
          })
          .map(window.mapActions)
      })
      expect(actual).toHaveLength(0)
      expect(actual).toMatchSnapshot()
    })

    test('partially above the fold', async () => {
      const actual = await page.evaluate(() => {
        window.scrollTo(0, window.innerHeight + 50)
        return window
          .computeScrollIntoView(document.querySelector('.vertical-scroll .target'), {
            scrollMode: 'if-needed',
          })
          .map(window.mapActions)
      })
      expect(actual).toHaveLength(1)
      expect(actual).toMatchSnapshot()
    })

    test('completely above the fold', async () => {
      const actual = await page.evaluate(() => {
        window.scrollTo(0, window.innerHeight + 100)
        return window
          .computeScrollIntoView(document.querySelector('.vertical-scroll .target'), {
            scrollMode: 'if-needed',
          })
          .map(window.mapActions)
      })
      expect(actual).toHaveLength(1)
      expect(actual).toMatchSnapshot()
    })
  })

  describe('horizontal', () => {
    test('completely overflowing', async () => {
      const actual = await page.evaluate(() => {
        window.scrollTo(0, 0)
        return window
          .computeScrollIntoView(document.querySelector('.horizontal-scroll .target'), {
            scrollMode: 'if-needed',
          })
          .map(window.mapActions)
      })
      expect(actual).toHaveLength(1)
      expect(actual).toMatchSnapshot()
    })

    test('partially overflowing', async () => {
      const actual = await page.evaluate(() => {
        window.scrollTo(50, 0)
        return window
          .computeScrollIntoView(document.querySelector('.horizontal-scroll .target'), {
            scrollMode: 'if-needed',
          })
          .map(window.mapActions)
      })
      expect(actual).toHaveLength(1)
      expect(actual).toMatchSnapshot()
    })

    test('completely in view', async () => {
      const actual = await page.evaluate(() => {
        window.scrollTo(window.innerWidth / 2, 0);
        return window
          .computeScrollIntoView(document.querySelector('.horizontal-scroll .target'), {
            scrollMode: 'if-needed',
          })
          .map(window.mapActions)
      })
      expect(actual).toHaveLength(0)
      expect(actual).toMatchSnapshot()
    })

    test('partially negative overflowing', async () => {
      const actual = await page.evaluate(() => {
        window.scrollTo(window.innerWidth + 50, 0)
        return window
          .computeScrollIntoView(document.querySelector('.horizontal-scroll .target'), {
            scrollMode: 'if-needed',
          })
          .map(window.mapActions)
      })
      expect(actual).toHaveLength(1)
      expect(actual).toMatchSnapshot()
    })

    test('fully negative overflowing', async () => {
      const actual = await page.evaluate(() => {
        window.scrollTo(window.innerWidth + 100, 0)
        return window
          .computeScrollIntoView(document.querySelector('.horizontal-scroll .target'), {
            scrollMode: 'if-needed',
          })
          .map(window.mapActions)
      })
      expect(actual).toHaveLength(1)
      expect(actual).toMatchSnapshot()
    })
  })
})
