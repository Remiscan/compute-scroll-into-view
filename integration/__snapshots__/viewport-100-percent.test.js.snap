// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scrollMode: if-needed (outside the scrollingElement bounding box) horizontal completely in view 1`] = `[]`;

exports[`scrollMode: if-needed (outside the scrollingElement bounding box) horizontal completely overflowing 1`] = `
[
  {
    "el": "html",
    "left": 100,
    "top": 0,
  },
]
`;

exports[`scrollMode: if-needed (outside the scrollingElement bounding box) horizontal fully negative overflowing 1`] = `
[
  {
    "el": "html",
    "left": 800,
    "top": 0,
  },
]
`;

exports[`scrollMode: if-needed (outside the scrollingElement bounding box) horizontal partially negative overflowing 1`] = `
[
  {
    "el": "html",
    "left": 800,
    "top": 0,
  },
]
`;

exports[`scrollMode: if-needed (outside the scrollingElement bounding box) horizontal partially overflowing 1`] = `
[
  {
    "el": "html",
    "left": 100,
    "top": 0,
  },
]
`;

exports[`scrollMode: if-needed (outside the scrollingElement bounding box) vertical completely above the fold 1`] = `
[
  {
    "el": "html",
    "left": 0,
    "top": 350,
  },
]
`;

exports[`scrollMode: if-needed (outside the scrollingElement bounding box) vertical completely below the fold 1`] = `
[
  {
    "el": "html",
    "left": 0,
    "top": 350,
  },
]
`;

exports[`scrollMode: if-needed (outside the scrollingElement bounding box) vertical completely in view 1`] = `[]`;

exports[`scrollMode: if-needed (outside the scrollingElement bounding box) vertical partially above the fold 1`] = `
[
  {
    "el": "html",
    "left": 0,
    "top": 350,
  },
]
`;

exports[`scrollMode: if-needed (outside the scrollingElement bounding box) vertical partially below the fold 1`] = `
[
  {
    "el": "html",
    "left": 0,
    "top": 350,
  },
]
`;
